### ISSUE TITLE
Regarding the depth scale

### ISSUE THREAD

**Issue by SunghwanHong** [crockwell/rel_pose#8]
Hi,
thanks for the nice work. 

I have one question regarding the depth scale in matterport. What I understood is that because matterport has rather larger translations than other datasets, the loss scale of translation error will be much bigger, which might affect negatively to the learning process. 
That's why in here:
https://github.com/crockwell/rel_pose/blob/35d1352fea09c941cef59b5f153ef3483db10bb5/src/data_readers/matterport.py#L48

you divided by 5 to adjust the scale?

If this is right, then my other question is, don't we have to change the intrinsics similarly? 

Thanks!

**Comment by crockwell** [comment]
Hello -- good question! Yeah, we divide by 5 to balance rotation and translation losses. However, you are correct, in theory one should adjust intrinsics for this. In our case, intrinsics are only learned to adjust positional encodings for the learned network; and we do not train or evaluate with varying intrinsics, i.e. we do not attempt to generalize from Matterport to other data. So not adjusting intrinsics shouldn't impact results. But to your point, yes, if one is using this in the future with the hope of generalizing to other intrinsics, I'd recommend scaling lambda for translation instead, and leaving DEPTH_SCALE=1. I'll leave it as-is for consistency with my experiments, but thanks for pointing this out!

**Comment by SunghwanHong** [comment]
Thanks for the crystal clear answer!

**Comment by SunghwanHong** [comment]
Hi, I was trying to compare our method with this paper, but I got one question. In the paper, you report the scores for SuperGlue.
Are these scores obtained using SuperPoint descriptors and keypoints?
In other words, are the scores actually SuperPoint + SuperGlue?

**Comment by crockwell** [comment]
Hi -- good question. Yep, it is indeed SuperPoint + SuperGlue.

**Comment by SunghwanHong** [comment]
Thanks! 

I tried to reproduce using my custom dataset, but the results were consistently worse than SP+SG. So I have been trying to debug and still working on it.

In the meanwhile, this might be a stupid question but from your experiments, did your training loss and validation loss looked like this ?
![image](https://github.com/crockwell/rel_pose/assets/85327219/696b7499-267b-42b1-aa07-6d18047fd1a2)

I found it weird that training loss rapidly goes down, but validation remains wiggly..

**Comment by crockwell** [comment]
Hmm that is frustrating. We actually didn't retrain SuperPoint / SuperGlue, but rather reported the results from [Sparse Planes](https://arxiv.org/abs/2103.14644) which just use the pretrained ScanNet model.

**Comment by SunghwanHong** [comment]
Even though I also tried to evaluate using the pretrained weights of your model, I obtained similar results to the reproduced results. I'm actually quite lost whether it's my fault or it's actually SP+SG was measured incorrectly..

**Comment by SunghwanHong** [comment]
I'll keep working on it and let you know when i resolve the issue. 
Just one last question is that when you were doing the experiments, did you see any overfitting problem like the graph above?

**Comment by crockwell** [comment]
I don't quite follow your first question. But sounds good. I only looked at curves for my method, and yeah, it definitely overfits. Although val curves still tend to decrease over training.

**Comment by SunghwanHong** [comment]
Actually, val curve issue should be fine, since I only used small subset. 
So just to make it clear, you also observed your curves also overfit?
If this is right, then I can assume I'm on the right direction.
